src/download.ts

```diff
@@ -36,6 +37,32 @@ export type CustomHnsDownloadOptions = CustomDownloadOptions & {
   hnsSubdomain?: string;
 };
 
+/**
+ * The response for a get file content request.
+ *
+ * @property data - The returned file content. Its type is stored in contentType.
+ * @property contentType - The type of the content.
+ * @property metadata - The metadata in JSON format.
+ * @property skylink - 46-character skylink.
+ */
+export type GetFileContentResponse<T = unknown> = {
+  data: T;
+  contentType: string;
+  metadata: Record<string, unknown>;
+  skylink: string;
+};
+
+/**
+ * The response for a get metadata request.
+ *
+ * @property metadata - The metadata in JSON format.
+ * @property skylink - 46-character skylink.
+ */
+export type GetMetadataResponse = {
+  metadata: Record<string, unknown>;
+  skylink: string;
+};
```

Comment:
Should we combine these return types with `data?: T` ? contentType is returned on HEAD calls and could serve a purpose.

Reply:
I want to keep the response types separate, but I added contentType to the metadata response. I'm can see potential use cases for it